frontend/src/ui/SearchableSelect.tsx

```diff
@@ -144,33 +150,52 @@ export const SeriesSelector: React.FC<SeriesSelectorProps> = ({
         }
     `;
 
-    const load = (input: string, callback: (options: readonly SeriesOption[]) => void) => {
-        fetchQuery<SearchableSelectSeriesQuery>(environment, query, { q: input, writableOnly })
-            .subscribe({
-                next: ({ series }) => {
-                    if (series.items === undefined) {
-                        setError(t("search.unavailable"));
-                        return;
-                    }
-
-                    callback(series.items.map(item => ({
-                        ...item,
-                        // Series returned by the search API have a different ID
-                        // prefix than other series. And the mutation expects an ID
-                        // starting with `ev`.
-                        id: item.id.replace(/^ss/, "sr"),
-                    })));
-                },
-                start: () => {},
-                error: (error: Error) => setError(<ErrorDisplay error={error} />),
-            });
+    const playlistsQuery = graphql`
+        query SearchableSelectPlaylistsQuery($q: String!, $writableOnly: Boolean!) {
+            playlists: searchAllPlaylists(query: $q, writableOnly: $writableOnly) {
+                ... on PlaylistSearchResults {
+                    items { id opencastId title description }
+                }
+            }
+        }
+    `;
+
+    type QueryResponse = SearchableSelectSeriesQuery$data | SearchableSelectPlaylistsQuery$data;
+
+    const createLoadFunction = (
+        query: GraphQLTaggedNode,
+        prefix: "sr" | "pl",
+    ) => (
+        input: string,
+        callback: (options: readonly VideoListOption[]) => void,
+    ) => {
+        fetchQuery(environment, query, { q: input, writableOnly }).subscribe({
+            next: response => {
+                const items = Object.values(response as QueryResponse)[0].items;
+
+                if (items === undefined) {
+                    setError(t("search.unavailable"));
+                    return;
+                }
+
+                callback(items.map(item => ({
+                    ...item,
+                    id: item.id.replace(/^../, prefix),
+                })));
+            },
+            start: () => {},
+            error: (error: Error) => setError(<ErrorDisplay error={error} />),
+        });
     };
 
+    const loadSeries = createLoadFunction(seriesQuery, "sr");
+    const loadPlaylists = createLoadFunction(playlistsQuery, "pl");
+
```

Comment:
Thats a good way to do it. I would indent the inner function once so make the formatting easier to read (IMO). I also tried to get rid of the `Object.values` thing there. I succeeded but I'm not entirely sure if it's better. Still not perfectly type safe

EDIT: what the heck, eslint is very annoying with indentation here... maybe you can figure it out. No need to reconfigure eslint, just do what it says, it's fine :D 

```suggestion
const makeLoader = <Q extends SearchableSelectSeriesQuery | SearchableSelectPlaylistsQuery>(
        query: GraphQLTaggedNode,
        prefix: "sr" | "pl",
    ) => (
        (
            input: string,
            callback: (options: readonly VideoListOption[]) => void,
        ) => {
            fetchQuery<Q>(environment, query, { q: input, writableOnly }).subscribe({
                next: r => {
                    const items = ("series" in r ? r.series : r.playlists).items;

                    if (items === undefined) {
                        setError(t("search.unavailable"));
                        return;
                    }

                    callback(items.map(item => ({
                        ...item,
                        id: item.id.replace(/^../, prefix),
                    })));
                },
                start: () => {},
                error: (error: Error) => setError(<ErrorDisplay error={error} />),
            });
        }
    );

    const loadSeries = makeLoader<SearchableSelectSeriesQuery>(seriesQuery, "sr");
    const loadPlaylists = makeLoader<SearchableSelectPlaylistsQuery>(playlistsQuery, "pl");
```

Reply:
Oh neat, I swear I was really close to that and was sure generics would be the way to go but something just wouldn't work. This however does.